website/static/js/comment.js

```diff
@@ -291,6 +511,50 @@ var CommentModel = function(data, $parent, $root) {
         self.author = self.$root.author;
     }
 
+    self.editableContent = ko.computed(function() {
+        var content = self.content();
+        if (!content) {
+            content = '';
+        }
+        var regex = /\[(@|\+)(.*?)\]\(\/([a-z\d]{5})\/\)/;
+        var matches = content.match(/\[(@|\+)(.*?)\]\(\/([a-z\d]{5})\/\)/g);
```

Comment:
>
* Why do we need a separate RegEx without the g flag (the regex variable here)? Wouldn't the g version also work for the second regex match in our algorithm here (the one inside the loop)?
* Do we really need two RegExes here? We can't make this work with a single RegEx?
>

I had a problem where the `regex` with the `g` flag would only match on every second attempt, as described in this [stackoverflow](http://stackoverflow.com/questions/1520800/why-regexp-with-global-flag-in-javascript-give-wrong-results), so I decided to use remove the `g` flag for the loop. The other option would be to manually reset the `lastIndex` for the `RegExp` object but I had some trouble when I attempted that solution. I could try again though.